nemo/collections/nlp/modules/common/megatron/attention.py

```diff
@@ -911,3 +930,84 @@ def forward(
         context_layer = context_layer.view(*new_context_layer_shape)
 
         return context_layer
+
+    def reset_fa_causal(self, query_length, key_length, causal):
```

Comment:
Unless I'm misunderstanding this function is doing more of an "is_causal" check rather than a reset of the property right? Can we rename it to reflect this if so?

Reply:
changed to `reset_is_causal()`